moabb/pipelines/classification.py

```diff
@@ -511,3 +532,124 @@ def predict_proba(self, X):
             y_pred[trial_n] = rho
 
         return y_pred
+
+
+class SSVEP_MsetCCA(BaseEstimator, ClassifierMixin):
+    """Classifier based on MsetCCA for SSVEP
+
+     The MsetCCA method learns multiple linear transforms to extract
+     SSVEP common features from multiple sets of EEG data. These are then used
+     to compute the reference signal used in CCA [1]_.
+
+    Parameters
+    ----------
+    freqs : dict with n_classes keys
+        Frequencies corresponding to the SSVEP stimulation frequencies.
+        They are used to identify SSVEP classes presents in the data.
+
+    n_filters: int
+        Number of multisets spatial filters used per sample data.
+        It corresponds to the number of eigen vectors taken the solution of the
+        MAXVAR objective function as formulated in Eq.5 in [1]_.
+
+    n_components: int
+        Number of CCA components to be used.
+
+
+    References
+    ----------
+
+    .. [1] Zhang, Y.U., Zhou, G., Jin, J., Wang, X. and Cichocki, A. (2014). Frequency
+           recognition in SSVEP-based BCI using multiset canonical correlation analysis.
+           International journal of neural systems, 24(04), p.1450013.
+           https://doi.org/10.1142/S0129065714500130
+    """
+
+    def __init__(self, freqs, n_filters=1, n_components=1):
+        self.n_filters = n_filters
+        self.freqs = freqs
+        self.n_components = n_components
+        self.cca = CCA(n_components=n_components)
+
+    def fit(self, X, y, sample_weight=None):
+        """
+        Compute the optimized reference signal at each stimulus frequency
+        """
+        self.classes_ = np.unique(y)
+        self.one_hot = {}
+        for i, k in enumerate(self.classes_):
+            self.one_hot[k] = i
+        n_trials, n_channels, n_times = X.shape
+        # Whiten signal X
+        X_white = np.zeros_like(X)
+        for trial in range(n_trials):
+            X_white[trial, :, :] = X[trial, :, :]
+            X_white[trial, :, :] = X_white[trial, :, :] - np.mean(
+                X_white[trial, :, :], axis=1, keepdims=True
+            )
+            C = np.cov(X_white[trial, :, :])
+            eig_val, eig_vec = linalg.eigh(C)
+            V = linalg.sqrtm(linalg.inv(np.diag(eig_val + 1e-10))) @ eig_vec.T
```

Comment:
it could be better to use shrinkage estimator rather than using cov + adding an arbitrary scale factor on the eigenvalues.